tfjs-backend-webgpu/src/binary_op_util.ts

```diff
@@ -216,6 +217,59 @@ function getBinaryWithNanString(op: string, useVec4: boolean) {
 
 export function getBinaryOpString(
     type: BinaryOpType, useVec4?: boolean): string {
+  let doOpSnippet: string;
+
+  // NaN-sensitive ops
+  do {
+    switch (type) {
+      case BinaryOpType.ATAN2:
+        if (!useVec4) {
+          continue;
+        }
+        doOpSnippet = getBinaryWithNanString('atan2', true);
+        break;
+      case BinaryOpType.MAX:
+        if (!useVec4) {
+          continue;
+        }
+        doOpSnippet = getBinaryWithNanString('max', true);
+        break;
+      case BinaryOpType.MIN:
+        if (!useVec4) {
+          continue;
+        }
+        doOpSnippet = getBinaryWithNanString('min', true);
+        break;
+      case BinaryOpType.MOD:
+        if (!useVec4) {
+          continue;
+        }
+        doOpSnippet = MOD_VEC4;
+        break;
+      case BinaryOpType.NOT_EQUAL:
+        if (!useVec4) {
+          continue;
+        }
+        doOpSnippet = NOT_EQUAL_VEC4;
+        break;
+      case BinaryOpType.POW:
+        if (!useVec4) {
+          continue;
+        }
+        doOpSnippet = POW_VEC4;
+        break;
+      default:
+        continue;
+    }
+    const checkNaNSnippet =
+        useVec4 ? CHECK_NAN_SNIPPET_VEC4 : CHECK_NAN_SNIPPET;
+    return `
+      ${doOpSnippet}
+      ${checkNaNSnippet}
```

Comment:
This introduces a duplicated check that is not necessary.

Reply:
_"That's right. It's part of the plan." -- The Dark Forest_